admin/restore.php

```diff
@@ -111,15 +114,20 @@ function confirmSubmit()
 			print "<script>javascript:document.body.style.cursor = 'default';</script>";
 		} 
 	}
-	$fejl = $_FILES['uploadedfile']['error'];
+	// if_isset cannot be used here since 'error' will be 0 on succes. But if_isset would return it as false
+	if (isset($_FILES['uploadedfile']['error'])) {
+		$fejl = $_FILES['uploadedfile']['error'];
+	} else {
+			$fejl = false;
+	}
 	if ($fejl) {
 		print "<script>javascript:document.body.style.cursor = 'default';</script>";
 		switch ($fejl) {
 			case 1: print "<BODY onLoad=\"javascript:alert('Filen er for stor - Kontroller upload_max_filesize i php.ini')\">";
 			case 2: print "<BODY onLoad=\"javascript:alert('Filen er for stor - er det en SALDI-sikkerhedskopi?')\">";
 		}
 	}
-	if (basename($_FILES['uploadedfile']['name'])) {
+	if (isset($_FILES['uploadedfile']['name']) && basename($_FILES['uploadedfile']['name'])) {
```

Comment:
Is the basename check needed?
Like #31 this is a boolean comparison on a [string function](https://www.php.net/manual/en/function.basename.php) which can lead to [unexpected behavior](https://www.php.net/manual/en/language.types.boolean.php#language.types.boolean.casting)

Reply:
I think your observation is correct, we need to make sure the file exsists and has a valid file, maybe we should use an if(!empty()) on it.